template.md

```diff
@@ -0,0 +1,321 @@
+
+- Put a backslash before the special character to "escape" it (for example, `\*` for a \*)
+- Use the [HTML entity code](http://www.ascii.cl/htmlcodes.htm) for the character (for example, `&#42;` for a &#42;).
+
+## File name
+
+File names use the following rules:
+* Contain only lowercase letters, numbers, and hyphens.
+* No spaces or punctuation characters. Use the hyphens to separate words and numbers in the file name.
+* Use action verbs that are specific, such as develop, buy, build, troubleshoot. No -ing words.
+* No small words - don't include a, and, the, in, or, etc.
+* Must be in Markdown and use the .md file extension.
+
+## Headings
+
+Use sentence-style capitalization. Always capitalize:
+- The first word of a heading. 
+- The word following a colon in a title or heading (for example, "How to: Sort an array"). 
+
+Headings should be done using atx-style, that is, use 1-6 hash characters (#) at the start of the line to indicate a heading, corresponding to HTML headings levels H1 through H6. Examples of first- and second-level headers are used above. 
+
+There **must** be only one first-level heading (H1) in your topic, which will be displayed as the on-page title.
+
+If your heading finishes with a `#` character, you need to add an extra `#` character in the end in order for the title to render correctly. For example, `# Async Programming in F# #`.     
+
+Second-level headings will generate the on-page TOC that appears in the "In this article" section underneath the on-page title.
+
+### Third-level heading
+#### Fourth-level heading
+##### Fifth level heading
+###### Sixth-level heading
+ 
+## Text styling
+
+*Italics*
+ Use for files, folders, paths (for long items, split onto their own line) - new terms - URLs (unless rendered as links, which is the default).
+
+**Bold**
+Use for UI elements.
+
+~~Strikethrough~~
+
+## Links
+
+### Internal Links
+
+To link to a header in the same Markdown file, view the source of the published article, find the id of the head (for example, `id="blockquote"`), and link using # + id (for example, `#blockquote`).
+
+- Example: [Blockquotes](#blockquote)
+
+To link to a Markdown file in the same repo, use [relative links](https://www.w3.org/TR/WD-html40-970917/htmlweb.html#h-5.1.2), including the ".md" at the end of the filename.
+
+- Example: [Readme file](readme.md)
+- Example: [Welcome to .NET](./docs/welcome.md)
+
+To link to a header in a Markdown file in the same repo, use relative linking + hashtag linking.
+
+- Example: [.NET Community](./docs/welcome.md#community)
+
+### External Links
+
+To link to an external file, use the full URL as the link.
+
+- Example: [GitHub](http://www.github.com)
+
+If a URL appears in a Markdown file, it will be transformed into a clickable link.
+
+- Example: http://www.github.com
+
+### Links to APIs
+
+Our build system has some extensions that are compatible with GFM and allow us to link to .NET Core APIs without having to use external links.  
```

Comment:
I missed that, no need to redefine everywhere, but I wonder if the mention is even necessary here.  I think you could drop "that are compatible with GFM and" since they don't actually work when rendered on GitHub.